### PR TITLE
enable xwayland and drm for weston on qemu-amd64

### PR DESCRIPTION
**Pull request by benwaffle**

### CODE REVIEW

**Comment by ollieparanoid** [comment]
Could you also increase the `pkgrel` here? Thank you very much for your contribution!

**Comment by benwaffle** [comment]
updated

**Comment by PabloCastellano** [comment]
I feel that this change is good, but I don't really know what are the advantages of it. Could you explain? :smiley_cat:

File: aports/main/postmarketos-ui-weston/APKBUILD [hunk 1]
```diff
@@ -1,6 +1,6 @@
 pkgname=postmarketos-ui-weston
 pkgver=1
-pkgrel=3
+pkgrel=4
```
Comment (PabloCastellano) [inline aports/main/postmarketos-ui-weston/APKBUILD]: You should bump `device-qemu-amd64/APKBUILD` instead of this

**Comment by PabloCastellano** [comment]
@PureTryOut You should try this. See this `weston.log`

```
[21:33:29.545] Command line: weston
[21:33:29.545] OS: Linux, 4.13.0, #2-postmarketOS PREEMPT Tue Sep 5 20:46:08 GMT 2017, x86_64
[21:33:29.546] Using config file '/etc/xdg/weston/weston.ini'
[21:33:29.547] Output repaint window is 7 ms maximum.
[21:33:29.547] Loading module '/usr/lib/libweston-3/drm-backend.so'
[21:33:29.571] initializing drm backend
[21:33:29.571] using /dev/dri/card0
[21:33:29.571] Loading module '/usr/lib/libweston-3/gl-renderer.so'
pci id for fd 12: 1234:1111, driver (null)
[21:33:30.451] EGL client extensions: EGL_EXT_client_extensions
               EGL_EXT_platform_base EGL_KHR_client_get_all_proc_addresses
               EGL_KHR_debug EGL_EXT_platform_wayland EGL_EXT_platform_x11
               EGL_MESA_platform_gbm
[21:33:30.452] warning: neither EGL_EXT_swap_buffers_with_damage or EGL_KHR_swap_buffers_with_damage is supported. Performance could be affected.
[21:33:30.452] EGL_KHR_surfaceless_context available
[21:33:30.489] EGL version: 1.4 (DRI2)
[21:33:30.489] EGL vendor: Mesa Project
[21:33:30.489] EGL client APIs: OpenGL OpenGL_ES 
[21:33:30.489] EGL extensions: EGL_EXT_buffer_age EGL_KHR_cl_event2
               EGL_KHR_config_attribs EGL_KHR_create_context
               EGL_KHR_fence_sync EGL_KHR_get_all_proc_addresses
               EGL_KHR_gl_renderbuffer_image EGL_KHR_gl_texture_2D_image
               EGL_KHR_gl_texture_cubemap_image EGL_KHR_image
               EGL_KHR_image_base EGL_KHR_image_pixmap
               EGL_KHR_no_config_context EGL_KHR_reusable_sync
               EGL_KHR_surfaceless_context EGL_KHR_wait_sync
               EGL_MESA_configless_context EGL_MESA_image_dma_buf_export
[21:33:30.489] GL version: OpenGL ES 3.0 Mesa 17.1.4
[21:33:30.489] GLSL version: OpenGL ES GLSL ES 3.00
[21:33:30.489] GL vendor: VMware, Inc.
[21:33:30.489] GL renderer: Gallium 0.4 on llvmpipe (LLVM 4.0, 128 bits)
[21:33:30.489] GL extensions: GL_EXT_blend_minmax GL_EXT_multi_draw_arrays
               GL_EXT_texture_format_BGRA8888
               GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth24
               GL_OES_element_index_uint GL_OES_fbo_render_mipmap
               GL_OES_mapbuffer GL_OES_rgb8_rgba8 GL_OES_standard_derivatives
               GL_OES_stencil8 GL_OES_texture_3D GL_OES_texture_float
               GL_OES_texture_float_linear GL_OES_texture_half_float
               GL_OES_texture_half_float_linear GL_OES_texture_npot
               GL_OES_vertex_half_float GL_EXT_texture_sRGB_decode
               GL_OES_EGL_image GL_OES_depth_texture
               GL_OES_packed_depth_stencil GL_EXT_texture_type_2_10_10_10_REV
               GL_OES_get_program_binary GL_APPLE_texture_max_level
               GL_EXT_discard_framebuffer GL_EXT_read_format_bgra
               GL_EXT_frag_depth GL_NV_fbo_color_attachments
               GL_OES_EGL_image_external GL_OES_EGL_sync
               GL_OES_vertex_array_object GL_EXT_texture_rg
               GL_EXT_unpack_subimage GL_NV_draw_buffers GL_NV_read_buffer
               GL_NV_read_depth GL_NV_read_depth_stencil GL_NV_read_stencil
               GL_EXT_draw_buffers GL_EXT_map_buffer_range GL_KHR_debug
               GL_OES_depth_texture_cube_map GL_OES_surfaceless_context
               GL_EXT_color_buffer_float GL_EXT_separate_shader_objects
               GL_EXT_shader_integer_mix GL_EXT_base_instance
               GL_EXT_compressed_ETC1_RGB8_sub_texture GL_EXT_copy_image
               GL_EXT_draw_buffers_indexed GL_EXT_draw_elements_base_vertex
               GL_EXT_polygon_offset_clamp GL_EXT_texture_border_clamp
               GL_KHR_context_flush_control GL_OES_copy_image
               GL_OES_draw_buffers_indexed GL_OES_draw_elements_base_vertex
               GL_OES_texture_border_clamp GL_OES_texture_stencil8
               GL_EXT_blend_func_extended GL_EXT_clip_cull_distance
               GL_MESA_shader_integer_functions
[21:33:30.489] GL ES 2 renderer features:
               read-back format: BGRA
               wl_shm sub-image to texture: yes
               EGL Wayland extension: no
[21:33:30.489] event0  - [21:33:30.489] Power Button: [21:33:30.489] is tagged by udev as: Keyboard
[21:33:30.489] event0  - [21:33:30.489] Power Button: [21:33:30.489] device is a keyboard
[21:33:30.489] event1  - [21:33:30.489] AT Translated Set 2 keyboard: [21:33:30.489] is tagged by udev as: Keyboard
[21:33:30.489] event1  - [21:33:30.489] AT Translated Set 2 keyboard: [21:33:30.489] device is a keyboard
[21:33:30.490] event2  - [21:33:30.490] ImExPS/2 Generic Explorer Mouse: [21:33:30.490] is tagged by udev as: Mouse
[21:33:30.490] event2  - [21:33:30.490] ImExPS/2 Generic Explorer Mouse: [21:33:30.490] device is a pointer
[21:33:30.516] Registered plugin API 'weston_drm_output_api_v1' of size 24
[21:33:30.516] Chosen EGL config details:
               RGBA bits: 8 8 8 0
               swap interval range: 0 - 0
[21:33:30.516] addfb2 failed: Invalid argument
[21:33:30.516] failed to create kms fb: Invalid argument
[21:33:30.516] cursor buffers unavailable, using gl cursors
[21:33:30.516] Failed to initialize backlight
[21:33:30.516] Output Virtual-1, (connector 29, crtc 27)
               mode 1024x768@60.0, preferred, current
               mode 1920x1080@60.0
               mode 1600x1200@60.0
               mode 1680x1050@60.0
               mode 1680x1050@59.9
               mode 1400x1050@60.0
               mode 1400x1050@59.9
               mode 1600x900@60.0
               mode 1280x1024@60.0
               mode 1440x900@59.9
               mode 1440x900@59.9
               mode 1280x960@60.0
               mode 1366x768@59.8
               mode 1366x768@60.0
               mode 1360x768@60.0
               mode 1280x800@59.8
               mode 1280x800@59.9
               mode 1280x768@59.9
               mode 1280x768@60.0
               mode 1280x720@60.0
               mode 800x600@60.3
               mode 800x600@56.2
               mode 848x480@60.0
               mode 640x480@59.9
[21:33:30.516] Compositor capabilities:
               arbitrary surface rotation: yes
               screen capture uses y-flip: yes
               presentation clock: CLOCK_MONOTONIC, id 1
               presentation clock resolution: 0.000000001 s
[21:33:30.516] Loading module '/usr/lib/weston/desktop-shell.so'
[21:33:30.521] launching '/usr/lib/weston/weston-keyboard'
[21:33:30.521] launching '/usr/lib/weston/weston-desktop-shell'
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
```

File: aports/main/postmarketos-ui-weston/APKBUILD (hunk 1 above)
Comment (benwaffle) [reply aports/main/postmarketos-ui-weston/APKBUILD]: why?

Comment (benwaffle) [reply aports/main/postmarketos-ui-weston/APKBUILD]: oops, fixed

**Comment by benwaffle** [comment]
@PabloCastellano the xwayland may not be necessary, not sure what the default is

Using drm-backend.so tells weston to use certain GPU features. For example the GPU can draw the cursor directly instead of weston drawing the cursor pixels to the framebuffer. There are a bunch of resources online that can explain it better than I can.

**Comment by PureTryOut** [comment]
Yup, seems good. I thought we needed to enable virtio-gpu or something related to get OpenGL support, but that seems to not be the case. I guess we can merge this then.

**Comment by ollieparanoid** [comment]
###### *@PabloCastellano sorry for having a typo in your nickname there, I swear there is no edit button ._.*

Thanks for this awesome PR @benwaffle, and thanks for your reviewing and testing @PabloCastellano and @PureTryOut! 👍

**Pull request merged**